src/User32.Desktop/User32+DESKTOP_ACCESS_MASK.cs

```diff
@@ -0,0 +1,268 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Diagnostics;
+
+    /// <content>
+    /// Contains the <see cref="DESKTOP_ACCESS_MASK"/> nested type.
+    /// </content>
+    public partial class User32
+    {
+        /// <summary>
+        /// The <see cref="Kernel32.ACCESS_MASK"/> type is a bitmask that specifies a set of access rights in the access mask of an access control entry.
+        /// /// This is a set of Specific Rights bitmask that defines access rights and security for desktop objects
+        /// </summary>
+        /// <remarks>
+        /// <see cref="Kernel32.ACCESS_MASK"/> is quite well described here: http://blogs.msdn.com/b/openspecification/archive/2010/04/01/about-the-access-mask-structure.aspx
+        /// Desktop security and access rights are quite well described here: https://msdn.microsoft.com/en-us/library/windows/desktop/ms687391(v=vs.85).aspx
+        /// </remarks>
+        public struct DESKTOP_ACCESS_MASK : IComparable, IFormattable, IComparable<DESKTOP_ACCESS_MASK>, IEquatable<DESKTOP_ACCESS_MASK>, IComparable<Kernel32.ACCESS_MASK>, IEquatable<Kernel32.ACCESS_MASK>
```

Comment:
Well as far as I understood ACCESS_MASK has some levels of "flagging" things and there is a specific part that other APIs can extend. This semantics is really interesting and the guy that inputted the Kernel32 port placed a really nice link that enlightened things for me.

On the Window Station API documentation, as well as on the Desktop API documentation, the specific flags semantics for these kinds of objects are explained in detail. I would just extend the Kernel32.ACCESS_MASK or the  Kernel32.ACCESS_MASK.SpecificRight if I could without much refactoring but Kernel32.ACCESS_MASK.SpecificRight is an enum and cant be extended, also Kernel32.ACCESS_MASK would need some refactoring to be able to have extensions coming from other packages. As far as I understood we will see this kind of things happen more and more when we start mapping the security and access functions more aggressively.

What I tried to do instead is make the least possible to make it compatible with the Kernel32.ACCESS_MASK struct, being convertible and comparable with it so that we could use any one. The main difference is on the SpecificRight flags and the fact the the GENERIC_* flags change semantics and meaning when applied on these kind of objects.